sale_commission_product_criteria/models/commission.py

```diff
@@ -0,0 +1,392 @@
+            WHERE
+                (item.product_tmpl_id IS NULL OR item.product_tmpl_id = any(%s))
+                AND (item.product_id IS NULL OR item.product_id = any(%s))
+                AND (item.categ_id IS NULL OR item.categ_id = any(%s))
+                AND (item.commission_id = %s)
+            ORDER BY
+                item.applied_on, categ.complete_name desc, item.id desc
+            """,
+            (
+                product.product_tmpl_id.ids,
+                product.ids,
+                categ_ids,
+                commission._origin.id,  # Added this
+            ),
+        )
+        item_ids = [x[0] for x in self.env.cr.fetchall()]
+        return item_ids
+
+    def _get_single_commission_amount(self, commission, subtotal, product, quantity):
+        self.ensure_one()
+        item_ids = self._get_commission_items(commission, product)
+        if not item_ids:
+            return 0.0
+        commission_item = self.env["commission.item"].browse(item_ids[0])
+        if commission.amount_base_type == "net_amount":
+            # If subtotal (sale_price * quantity) is less than
+            # standard_price * quantity, it means that we are selling at
+            # lower price than we bought, so set amount_base to 0
+            subtotal = max([0, subtotal - product.standard_price * quantity])
+        self.applied_commission_item_id = commission_item
+        # if self.agent_id.use_multi_type_commissions:
+        self.applied_commission_id = commission_item.commission_id
+        if commission_item.commission_type == "fixed":
+            return commission_item.fixed_amount
+        elif commission_item.commission_type == "percentage":
+            return subtotal * (commission_item.percent_amount / 100.0)
+
+
+class SaleCommission(models.Model):
+    _inherit = "sale.commission"
+
+    commission_type = fields.Selection(
+        selection_add=[("rules", "Rules")], ondelete={"rules": "cascade"}
+    )
+    item_ids = fields.One2many("commission.item", "commission_id")
+    sequence = fields.Integer(
+        "Sequence", default=1, help="The first in the sequence is the default one."
+    )
+
+    def write(self, vals):
+        res = super(SaleCommission, self).write(vals)
+        self.sort_items()
+        return res
+
+    def sort_items(self):
+        # Priority 1: product_variant > product > product_category > global
+        # Priority 2 (inside of priority 1):
+        # Discount > SOL
+        # Priority 3 (inside of priority 2):
+        # Regular with price list > Regular without price list
+        for com_type, seq in COMMISSION_ITEM_TYPES.items():
+            all_lines = self.item_ids.filtered(lambda x: x.applied_on == com_type)
+
+            sol_lines = all_lines.filtered(lambda x: x.based_on == "sol")
+            sol_lines.update({"sequence": seq + 100})
+
+    def action_archive(self):
+        res = super().action_archive()
+        self.item_ids.write({"active": False})
```

Comment:
Can you avoid this and simply in the SQL don't take into account inactive commission types?

Reply:
@pedrobaeza can you clarify?

Reply:
This is not needed according the latest SQL.